BrainPortal/app/controllers/nh_projects_controller.rb

```diff
@@ -84,12 +87,139 @@ def update #:nodoc:
+    @can_add_license = @nh_project.creator_id == current_user.id
   end
 
   def files #:nodoc:
     @nh_project = find_nh_project(current_user, params[:id])
     @files      = @nh_project.userfiles
   end
 
+  def new_license #:nodoc:
+    @nh_project = find_nh_project(current_user, params[:id] || params[:nh_project_id], false)
+
+    unless @nh_project.creator_id == current_user.id
+      cb_error "Only owner can set licensing", :redirect  => :neurohub
+    end
+  end
+
+  def add_license #:nodoc:
+    content = params[:meta][:license_description]
+    @nh_project = find_nh_project(current_user, params[:id] || params[:nh_project_id], false)
+    cb_error 'Presently empty licenses are not allowed' if content.blank?
+    unless @nh_project.creator_id == current_user.id
+      cb_error "Only owner can set licensing", :redirect  => :neurohub
+    end
+    @nh_project = find_nh_project(current_user, params[:id], false)
+    @nh_project.create_license_file(content, current_user)
+    flash['message'] = 'A license is added. You can force users to sign multiple license agreements if needed'
+    redirect_to :action => :show
+  end
+
+  def show_license #:nodoc:
+    @nh_project = find_nh_project(current_user, params[:id], false)
+    @unsigned_licenses = current_user.o_unsigned_custom_licenses(@nh_project)
+    if @unsigned_licenses.empty?
+      if @nh_project.license_agreements.present?
+        flash[:error] = 'You already signed all license'
+      else
+        flash[:error] = 'No license is defined for this project'
+        redirect_to :action => :show
+        return
+      end
+    end
+    @license = @unsigned_licenses[0]
+    @userfile = Userfile.find(@license)
+    set_viewer(@license)
+  end
+
+  def sign_license #:nodoc:
+    @nh_project = find_nh_project(current_user, params[:id], false)
+    @unsigned_licenses = current_user.o_unsigned_custom_licenses(@nh_project)
+    @license = params[:license]
+    unless @nh_project.license_agreements.include?(@license)
+      flash['error'] = 'You are trying to access unrelated license. Try again or report the issue to the support.'
+      redirect_to :action => :show
+      return
+    end
+    if current_user.all_custom_licenses_signed.include?(@license)
+      flash['error'] = 'You cannot sign the license because you have already signed it.'
+      redirect_to :action => :show
+      return
+    end
+    @userfile = Userfile.find(@license)
+    unless params.has_key?(:agree)
+      flash[:error] = "You cannot access that project without signing the End User Licence Agreement first."
+      redirect_to "/neurohub"
+      return
+    end
+
+    if params.keys.grep(/\Alicense_check/).size < 1
+        flash[:error] = "There was a problem with your submission. Please read the agreement and check all checkboxes."
+        redirect_to :action => :show
+        return
+    end
+
+    current_user.all_custom_licenses_signed = current_user.all_custom_licenses_signed << @license
```

Comment:
You can use `.push` is more `rubyist`

Reply:
Apparently not everybody think so. More, `<< ` is several times more used that `.push` so I prefer stick with majority. Yet if you are convinced in superiority of `push` and keep pushing please tell me I have no other strong reason reason to keep `<<` save uniformity and (minuscule) performance improvement.